### PR TITLE
Add missing unary plus and cleanup tests

### LINKED ISSUE #1888: No implementation of plus(x)
## Description

Some more spring cleaning to do.

The following Stan signatures currently dont have a Stan Math implementation:
```stan
int plus(int)
real plus(real)
vector plus(vector)
row_vector plus(row_vector)
matrix plus(matrix)
```

This is the named version of the unary + operator. So these two are the same
```stan
int i = +x;
int i = plus(x);
```

#### Current Version:
v3.2.0

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

Fixes #1888  by adding the unary plus. When adding tests for that I saw that we still have the _works_with_other_functions tests that are redundant after https://github.com/stan-dev/math/pull/1980 This advances https://github.com/stan-dev/math/issues/1886 but does not close it yet.

## Tests

Added same tests as for minus()

## Side Effects

/

## Release notes

Added missing implementations for the unary plus.

## Checklist

- [x] Math issue #1888

- [x] Copyright holder: Rok Češnovar, Univ. of Ljubljana

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -47,8 +47,6 @@ help:
 	@echo '      * rev -> prim'
 	@echo '      * fwd -> prim'
 	@echo '      * mix -> {rev, fwd, prim}'
-	@echo '      * within {prim, rev, fwd, mix}: mat -> arr -> scal'
-	@echo '      * only include {prim, rev, fwd, mix}/meta.hpp from the meta subfolders'
```
Comment (rok-cesnovar) [inline makefile:51]: This is unrelated, just removed it because its no longer true. We do not check for those subfolder as they no longer exist.

Comment (t4c1) [reply makefile:51]: I think the line 51 is still relevan (50 can be deleted)

File: stan/math/prim/fun/plus.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * Returns the unary plus of the input.
+ *
+ * @tparam T Type of input.
+ * @param x input.
+ * @return result of unary plus of the input.
+ */
+template <typename T>
+inline plain_type_t<T> plus(const T& x) {
```
Comment (t4c1) [inline stan/math/prim/fun/plus.hpp:17]: Use perfect forwarding when returning the argument.

**Review by t4c1 — CHANGES_REQUESTED** [review]

File: makefile (hunk 1 above)
Comment (rok-cesnovar) [reply makefile:51]: We no longer enforce that after the flatten was done. Its encouraged but not enforced.

File: stan/math/prim/fun/plus.hpp (hunk 2 above)
Comment (rok-cesnovar) [reply stan/math/prim/fun/plus.hpp:17]: Like this?
```cpp
template <typename T>
inline plain_type_t<T> plus(T&& x) {
  return std::forward(x);
}
```

File: makefile (hunk 1 above)
Comment (t4c1) [reply makefile:51]: Ok, this is fine.

File: stan/math/prim/fun/plus.hpp (hunk 2 above)
Comment (t4c1) [reply stan/math/prim/fun/plus.hpp:17]: Yes, also make the return type `T` - we don't need plus to eval its argument.

**Comment by rok-cesnovar** [comment]
@serban-nicusor-toptal can we try if `gelman-group-linux` can now run the `Clang-format` and `Linting & Doc checks` again? If we are unlucky with Linux AWS instances like we are now and there are none free, the jobs are all stalled.

If there is an issue with the machine, we can see if we can fix it.

**Comment by serban-nicusor-toptal** [comment]
Hey @rok-cesnovar the pool was enough, there were no issues with it not being able to find machines.
I either introduced a bug yesterday or there were some connectivity issues today, I've recreated the fleets now and everything is working fine.
Sorry for the waiting!

**Comment by rok-cesnovar** [comment]
@serban-nicusor-toptal thanks!

**Review by t4c1 — APPROVED** [review]
